editor/editor_file_system.cpp

```diff
@@ -2211,8 +2211,11 @@ void EditorFileSystem::reimport_file_with_custom_parameters(const String &p_file
 }
 
 void EditorFileSystem::_reimport_thread(uint32_t p_index, ImportThreadData *p_import_data) {
-	p_import_data->max_index = MAX(p_import_data->reimport_from + int(p_index), p_import_data->max_index);
-	_reimport_file(p_import_data->reimport_files[p_import_data->reimport_from + p_index].path);
+	int current_max = p_import_data->reimport_from + int(p_index);
+	p_import_data->max_index_lock.lock();
+	p_import_data->max_index = MAX(current_max, p_import_data->max_index);
+	p_import_data->max_index_lock.unlock();
```

Comment:
```suggestion
	{
		MutexLock lock(p_import_data->max_index_lock);
		p_import_data->max_index = MAX(current_max, p_import_data->max_index);
	}
```
I'd suggest using a mutex lock, it's cleaner and also more secure

Reply:
Ah, at the first place I was looking for something smimilar to lock guard, did not realize Godot has one! Many thanks!